azure_monitor/src/azure_monitor/sdk/auto_collection/live_metrics/manager.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+import threading
+import time
+
+from opentelemetry.context import attach, detach, set_value
+from opentelemetry.sdk.metrics.export import MetricsExportResult
+
+from azure_monitor.sdk.auto_collection import live_metrics
+from azure_monitor.sdk.auto_collection.live_metrics.exporter import (
+    LiveMetricsExporter,
+)
+from azure_monitor.sdk.auto_collection.live_metrics.sender import (
+    LiveMetricsSender,
+)
+
+FALLBACK_INTERVAL = 60
+PING_INTERVAL = 5
+POST_INTERVAL = 1
+MAIN_INTERVAL = 2
+
+
+class LiveMetricsManager(threading.Thread):
+    """Live Metrics Manager
+
+    It will start Live Metrics process when instantiated,
+    responsible for switching between ping and post actions.
+    """
+
+    daemon = True
+
+    def __init__(self, meter, instrumentation_key):
+        super().__init__()
+        self.thread_event = threading.Event()
+        self.interval = MAIN_INTERVAL
+        self._instrumentation_key = instrumentation_key
+        self._is_user_subscribed = False
+        self._meter = meter
+        self._post = None
+        self._ping = LiveMetricsPing(self._instrumentation_key)
+        self.start()
+
+    def run(self):
+        self.check_if_user_is_subscribed()
+        while not self.thread_event.wait(self.interval):
```

Comment:
Nit: `timeout` should be an [float](https://docs.python.org/3/library/threading.html#threading.Event.wait).